specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -0,0 +1,160 @@
+
+### SQS / SNS
+Lambda functions that are triggered by SQS (or SNS) accept an `event` input that may contain one ore more SQS / SNS messages in the `event.records` array. All message-related context information (including the `traceparent`) is encoded in the individual message attributes (if at all). We cannot (automatically) wrap the processing of the individual messages that are sent as a batch of messages with a single `event`. 
+
+Thus, in case that an SQS / SNS `event` contains **exactly one** SQS / SNS message, the agents must apply the following, messaging-specific retrieval of information. Otherwise, the agents should apply the [Generic Lambda Instrumentation](generic-lambda-instrumentation) as desribed above.
+
+With only one message in `event.records`, the agents can use the single SQS / SNS `record` to retrieve the `traceparent` and `tracestate` from `record.messageAttributes` and use it for starting the lambda transaction. 
+
+In addition the following fields should be set for Lambda functions triggered by SQS or SNS:
+Field | Value | SQS | SNS
+---   | ---   | --- | ---
+`faas.trigger.type` | `pubsub` | Constant value for message based triggers | "
+`faas.trigger.id` | e.g. `arn:aws:sqs:us-east-2:123456789012:my-queue` | Use `record.eventSourceArn` |Use `record.sns.topicArn`
+`faas.trigger.name` | e.g. `my-queue` | Parse the queue / topic name (6th element) from the ARN used for `faas.trigger.id`. | "
+`faas.trigger.account.id` | e.g. `12345678912` | Parse account segment (5th) from the ARN used for `faas.trigger.id` | " 
+`faas.trigger.region` | e.g. `us-east-1` |  Use `record.awsRegion` | Parse region segment (4th) from `record.sns.topicArn`
+`faas.trigger.request_id` | e.g. `someMessageId` | Use `record.messageId` | Use `record.sns.messageId`
+`context.message.queue` | e.g. `arn:aws:sqs:us-east-2:123456789012:my-queue` | Use `record.eventSourceArn` | Use `record.sns.topicArn`
+`context.message.age` | e.g. `3298` | Age of the message in milliseconds. `current_time` - `SentTimestamp`, if SentTimestamp is available. </br></br> For SQS, the timestamp can be retrieved from message attributes with key `SentTimestamp`. | Age of the message in milliseconds. `current_time` - `SentTimestamp`, if SentTimestamp is available. </br></br> For SNS, the timestamp can be retrieved from `record.sns.timestamp`.
+`context.message.body` |  | The message body. Should only be captured if body capturing is enabled in the configuration.</br></br> Use `record.body` | The message body. Should only be captured if body capturing is enabled in the configuration.</br></br> Use `record.sns.message`
+`context.message.headers` |  | The message attributes. Should only be captured, if capturing headers is enabled in the configuration.</br></br> Use `record.messageAttributes`| The message attributes. Should only be captured, if capturing headers is enabled in the configuration.</br></br> Use `record.sns.messageAttributes`
```

Comment:
I think these can lose the `context.` prefix. All the `faas` fields are also nested under `context.` so it's confusing to include it here.